src/System.Collections.NonGeneric/tests/Performance/Perf.ArrayList.cs

```diff
@@ -0,0 +1,476 @@
+            ArrayList elements = CreateArrayListOfInts(length);
+            var arrayList = new ArrayList(elements);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        arrayList.InsertRange(index, elements);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 5000)]
+        public void CopyTo()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            int size = 10000;
+            ArrayList elements = CreateArrayListOfInts(size);
+            int[] destination = new int[size];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        elements.CopyTo(destination, 0);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 10000)]
+        public void LastIndexOf()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            int size = 10000;
+            ArrayList elements = CreateArrayListOfInts(size);
+            int element = size / 2;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        elements.LastIndexOf(element);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 1000)]
+        public void Remove()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            ArrayList elements = CreateArrayListOfInts();
+            int element = elements.Count / 2;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        elements.Remove(element);
+                    }                    
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 10000)]
+        public void RemoveRange()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            var random = new Random();
+            int size = (int)Benchmark.InnerIterationCount;
+            ArrayList elements = new ArrayList(size);
+            for (int index = 0; index < size; index++)
+            {
+                elements.Add(random.Next());
```

Comment:
What's the purpose of the randomness here?  Are the actual values used somewhere?

Reply:
amm... yes I am fixed this too....